src/crypto/kdf/mod.rs

```diff
@@ -0,0 +1,30 @@
+//! Key derivation
+//!
+//! Multiple secret subkeys can be derived from a single master key. Given the master key and a key
+//! identifier, a subkey can be deterministically computed. However, given a subkey, an attacker
+//! cannot compute the master key nor any other subkeys.
+//!
+//! This API was introduced in libsodium 1.0.12
+//!
+//! # Example
+//!
+//! ```
+//! use sodiumoxide::crypto::kdf;
+//! use sodiumoxide::crypto::secretbox;
+//!
+//! const CONTEXT: [u8; 8] = *b"Examples";
+//!
+//! let key = kdf::gen_key();
+//!
+//! let mut key1 = secretbox::Key([0; secretbox::KEYBYTES]);
+//! kdf::derive_from_key(&mut key1.0[..], 1, CONTEXT, &key).unwrap();
```

Comment:
Will it generate correct `crypto::secretbox::SecretKey`s?

Reply:
Yes. Any array of the correct size is a valid key for secretbox